src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -106,18 +101,6 @@ static Tuple<JWK, JWSSigner> createJwkFromSettings(Settings settings) {
         }
     }
 
-    public static String padSecret(String signingKey, JWSAlgorithm jwsAlgorithm) {
-        int requiredSecretLength;
-        try {
-            requiredSecretLength = getMinRequiredSecretLength(jwsAlgorithm);
-        } catch (JOSEException e) {
-            throw new RuntimeException(e);
-        }
-        int requiredByteLength = ByteUtils.byteLength(requiredSecretLength);
-        // padding the signing key with 0s to meet the minimum required length
-        return StringUtils.rightPad(signingKey, requiredByteLength, "\0");
-    }
-
```

Comment:
Hi, @RyanL1997 regarding this part of code. It was added for the backwards compatibility. If user uses signing secret that's shorter then 64bytes nimbus would throw errors like that:
![image](https://github.com/opensearch-project/security/assets/128170204/58025bba-89ff-4c2a-951c-d24d7eba5d96)
Overall I think that it would be a good idea to force users to use longer, safer secret key. On the other hand, in previous discussion, we agreed that users should not be surprised by the sudden change. This padding would also allow for JWTs generated by node running cxf implementation, to get parsed by node running with nimbus implementation.

Reply:
Hi @MaciejMierzwa , thanks for the info! Yes, I think if this is the case, lets bring back the padding function. I'm on it.

Reply:
Hi, take a look at this: https://github.com/opensearch-project/security/commit/eb4533766aeabbe0be3838de7f7976364c562054. I did bring it back with slight modifications in a different branch. As I understand the config for signing secret is base64url encoded? Padding I added again is taking this into consideration. So it's reading config in base64 -> decoding it -> padding -> creating jwk for signing jwt.